Add support for checkboxes on third screen of the admin panel

diegoserranoa opened the pull request:
Pull values from the admin panel and use them in the SDK.

## Changes

- Support `add_comment_param_to_url`,`add_email_param_to_url` and `add_score_param_to_url`
- Update test

## Test

This PR should be tested using the checkboxes present in the 3rd screen of customizing the survey.

![Screen Shot 2019-11-26 at 5 56 02 PM](https://user-images.githubusercontent.com/1431421/69679436-1e5fc680-1076-11ea-964c-1866cfed99ad.png)

Depending on what is selected there, the URL should contain `wootric_email`, `wootric_score` and `wootric_comment`.

Here's a little table to help testing

| Combinations  | Email    | Score    | Comment  |
|---------------|----------|----------|----------|
| Combination 1 | Enabled  | Enabled  | Enabled  |
| Combination 2 | Enabled  | Enabled  | Disabled |
| Combination 3 | Enabled  | Disabled | Enabled  |
| Combination 4 | Enabled  | Disabled | Disabled |
| Combination 5 | Disabled | Enabled  | Enabled  |
| Combination 6 | Disabled | Enabled  | Disabled |
| Combination 7 | Disabled | Disabled | Enabled  |
| Combination 8 | Disabled | Disabled | Disabled |

- [x] Combination 1
- [x] Combination 2
- [x] Combination 3
- [x] Combination 4
- [x] Combination 5
- [x] Combination 6
- [x] Combination 7
- [x] Combination 8

- [x] All
- [x] Detractors
- [x] Passives
- [x] Promoters

diegoserranoa committed:
Add support for checkboxes on third screen of the admin panel

Pull values from the admin panel and use them in the SDK.

Changes:
- Support `add_comment_param_to_url`,`add_email_param_to_url` and `add_score_param_to_url`
- Update test

prabhatjha commented:
@diegoserranoa can you pls do Testflight thing again?

marinhero commented on WootricSDK.podspec:
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name     = 'WootricSDK'
-  s.version  = '0.15.0'
+  s.version  = '0.16.0'
```
Don't forget to update the Segment version as well!

marinhero commented:
This is not working for the "All respondants" case :(

marinhero reviewed (changes requested):
Make sure to add a test for the `ALL` cases since it's not working and the build is passing

diegoserranoa commented:
Thanks for the review @marinhero ! I found the issue and updated the PR. Left the new changes in a different commit so it's easier for you to verify what I did and that I added the tests.

marinhero reviewed (changes requested):
This configuration:

<img width="286" alt="Screen Shot 2019-12-03 at 2 34 43 PM" src="https://user-images.githubusercontent.com/316711/70095623-16ee6f00-15da-11ea-9bdf-297ed9208f61.png">

won't attach the feedback to the URL. Please take a look at the code and add a test for this case

diegoserranoa commented on WootricSDK/WootricSDKTests/WTRSettingsTests.m:
```diff
@@ -86,6 +86,330 @@ - (void)testFollowupQuestionForPromoter {
+  XCTAssertEqualObjects(thankYouURL, thankYouLinkURL);
+}
+
+- (void)testThankYouLinkWithEmailCommentForAllRespondents {
```
Here is the test!

diegoserranoa commented:
@marinhero I just test that case again and it is working. I also included a test for that case here: https://github.com/Wootric/WootricSDK-iOS/pull/71/files/3a7725169809c262eb7f627d7e781c4108671024#diff-b8b12993c12d37ed6c57d26f46db2c39R343

marinhero reviewed (approved):
The comments have been addressed! Thanks Diego. Rebase, merge and please verify that Segment is getting the latest version. Don't forget to remove the "Not supported" banner from Customize Survey!
